src/lsp-server.spec.ts

```diff
@@ -274,7 +274,19 @@ describe('completion', () => {
         expect(proposals).not.toBeNull();
         const completion = proposals!.items.find(completion => completion.label === 'getById');
         expect(completion).toBeDefined();
-        expect(completion!.textEdit).toBeUndefined();
+        expect(completion!.textEdit).toMatchObject({
+            newText: 'getById',
+            range: {
+                start: {
+                    character: 20,
+                    line: 6,
+                },
+                end: {
+                    character: 23,
+                    line: 6,
+                },
+            },
```

Comment:
Technically not wrong but it could use the "old school" insertText instead since that has smaller payload and should work the same in all cases I know (unless you know specific cases where TextEdit produces better result).

We have to consider the payload size because typescript can sometimes return 50MB worth of completions so it's worth to be as optimal as possible.

Reply:
Also related I guess... For import completions we have both `textEdit` and `insertText / insertTextFormat` where the latter ones should be redundant and just make the payload bigger. For example:

```json
    {
      "commitCharacters": [
        ".",
        ",",
        "("
      ],
      "data": {
        "entryNames": [
          {
            "data": {
              "exportName": "readConfigFile",
              "fileName": "/usr/local/workspace/github/typescript-language-server/node_modules/typescript/lib/typescript.d.ts",
              "moduleSpecifier": "typescript"
            },
            "name": "readConfigFile",
            "source": "typescript"
          }
        ],
        "file": "/usr/local/workspace/github/typescript-language-server/src/completion.ts",
        "line": 8,
        "offset": 17
      },
      "detail": "typescript",
      "filterText": "import { readConfigFile$1 } from 'typescript';",
      "insertText": "import { readConfigFile$1 } from 'typescript';",
      "insertTextFormat": 2,
      "kind": 3,
      "label": "readConfigFile",
      "sortText": "11",
      "textEdit": {
        "insert": {
          "end": {
            "character": 18,
            "line": 7
          },
          "start": {
            "character": 0,
            "line": 7
          }
        },
        "newText": "import { readConfigFile$1 } from 'typescript';",
        "replace": {
          "end": {
            "character": 18,
            "line": 7
          },
          "start": {
            "character": 0,
            "line": 7
          }
        }
      }
    }
```

Reply:
OK, I won't insist on using `insertText` over `textEdit` since that is inferior and can in theory produce worse results (when part of the prefix includes trigger (like punctuation) characters.

But we can instead get rid of `insertText` (not set it) when `textEdit` is set.